File: 1221.cpp
/*
	Author: Zoljargal Gantumur
	Runtime: 0 ms, faster than 100.00% of C++ online submissions
	Memory Usage: 6.4 MB, less than 12.45% of C++ online submissions
*/ 

class Solution {
public:
    int balancedStringSplit(string s) {
        int a = 0;
        int ans = 0;
        for(int i=0; i<s.size(); i++){
            if(s[i] == 'R'){
                ans++;
            }else{
                ans--;
            }
            if (ans==0){
                a++;
            }
        }
        return a;
    }
};

File: 1221.py
# Author: Zoljargal Gantumur
# Runtime: 28 ms, faster than 73.49% of Python3 online submissions
# Memory Usage: 13.8 MB, less than 69.80% of Python3 online submissions

class Solution:
    def balancedStringSplit(self, s: str) -> int:
        ans = 0
        c = 0
        for i in range(len(s)):
            if(s[i]=='R'):
                ans+=1
            else:
                ans-=1
            if ans==0:
                c+=1
        return c
